controllers/nginx/pkg/template/template.go

```diff
@@ -111,7 +111,13 @@ func (t *Template) Write(conf config.TemplateConfig, isValidTemplate func([]byte
 	cmd.Stdout = t.outCmdBuf
 	if err := cmd.Run(); err != nil {
 		glog.Warningf("unexpected error cleaning template: %v", err)
-		return t.tmplBuf.Bytes(), nil
+		content := t.tmplBuf.Bytes()
+		err = isValidTemplate(content)
```

Comment:
Why not just change the signature of this method to
```
Write(conf config.TemplateConfig) ([]byte, error)
```
and call it like 
```
if err := Write(); err != nil {
  isValidTemplate(templ)
}
```

?
Seems more straightforward than passing in a function and calling it everytime there's an error returned from this function